libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/adapter/oracle/OracleAlterTableFactory.java

```diff
@@ -51,20 +51,21 @@ public AlterTable generate() {
 
     @Override
     public AlterTable visitAlter_table_stmt(Alter_table_stmtContext ctx) {
-        String relation = OracleFromReferenceFactory.getRelation(ctx.relation_factor());
         AlterTable alterTable;
         if (ctx.alter_table_actions() != null) {
             List<AlterTableAction> actions = ctx.alter_table_actions().alter_table_action().stream()
                     .map(c -> new OracleAlterTableActionFactory(c).generate()).collect(Collectors.toList());
-            alterTable = new AlterTable(ctx, relation, actions);
+            alterTable =
```

Comment:
we may use a same RealtionFactor instead of generating it twice:
```java
RelationFactor relationFactor = OracleFromReferenceFactory.getRelationFactor(ctx.relation_factor());
if (...) {
    alterTable = new AlterTable(ctx, relationFactor);
} else {
    ....
}
```

Reply:
In fact, the creation will not be duplicated because the if and else branches are mutually exclusive. However, your suggestion is still valid as it makes the code more readable. The changes have been made.